packages/presentation/lib/app/app_navigation_rail.dart

```diff
@@ -0,0 +1,90 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:presentation/app/app_bloc.dart';
+import 'package:presentation/app/app_data.dart';
+import 'package:presentation/config/dimens/dimens.dart';
+import 'package:presentation/config/theme/app_colors.dart';
+import 'package:presentation/generated_localization/l10n.dart';
+import 'package:presentation/utils/image_path.dart';
+
+class AppNavigationRail extends StatefulWidget {
+  const AppNavigationRail({
+    required this.bloc,
+    required this.blocData,
+    super.key,
+  });
+
+  final AppBloc bloc;
+  final AppData blocData;
+
+  @override
+  State<AppNavigationRail> createState() => _AppNavigationRailState();
+}
+
+class _AppNavigationRailState extends State<AppNavigationRail> {
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      width: Dimens.size100,
+      decoration: const BoxDecoration(
+        border: Border(
+          right: BorderSide(
+            color: AppColorsDark.borderTabBar,
+            width: Dimens.size1,
+          ),
+        ),
+      ),
+      child: NavigationRail(
+        backgroundColor: AppColorsDark.primaryColorDark,
+        selectedIndex: widget.blocData.currentPageIndex,
+        onDestinationSelected: widget.bloc.onItemTapped,
+        destinations: [
+          NavigationRailDestination(
```

Comment:
create the subclass and substitute selected and unselected colours inside it. so, u'll pass only iconpath and label string into subclass and inside would be mapping into superclass fields